virtio-devices/src/vhost_user/blk.rs

```diff
@@ -409,7 +411,13 @@ impl Snapshottable for Blk {
     }
 
     fn snapshot(&mut self) -> std::result::Result<Snapshot, MigratableError> {
-        Snapshot::new_from_versioned_state(&self.id(), &self.state())
+        let snapshot = Snapshot::new_from_versioned_state(&self.id(), &self.state())?;
+
+        if self.migration_started {
+            self.shutdown();
+        }
```

Comment:
I wonder why we choose not do the same (call to `shutdown`) for normal snapshot? Are we expecting that the VM being snapshotted will continue to communicate with the backend?

Reply:
The difference is that a snapshot is taken as a reference of your VM at a certain point in time, like a backup. That means you might be running a VM, but at some point you want to save the current state without the need for shutting down the VM. Later on, after you shutdown your VM, you might want to start a new one based off the snapshot you took earlier. That's how snapshot is different than live migration, there's no concurrency in accessing the same socket since we're not trying to run two identical VMs at the same time.

Reply:
Makes sense. Thanks for the explanation.